apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -0,0 +1,657 @@
+    {ok, Nonce} = rpc(aec_conductor, next_nonce_for_account, [PubKey]),
+    {ok, SpendTx} =
+        aec_spend_tx:new(
+          #{sender => PubKey,
+            recipient => random_hash(),
+            amount => Amount,
+            fee => Fee,
+            nonce => Nonce}),
+    {ok, SignedTx} = rpc(aec_keys, sign, [SpendTx]),
+    post_tx(SignedTx),
+    {ok, [SignedTx]} = rpc(aec_tx_pool, peek, [infinity]), % same tx 
+    ok.
+
+%% TODO: POST broken tx (currently not handled in API Endpoint)
+
+%% changing of another account's balance is checked in pending_transactions test
+miner_balance(_Config) ->
+    {ok, 404, #{<<"reason">> := <<"Account not found">>}} = get_balance(),
+    BlocksToMine = 10,
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToMine),
+    {ok, Bal} = rpc(aec_conductor, get_miner_account_balance, []),
+    {ok, 200, #{<<"balance">> := Bal}} = get_balance(),  
+    {ok, PubKey} = rpc(aec_keys, pubkey, []),
+    {ok, 200, #{<<"balance">> := Bal}} = get_balance(base64:encode(PubKey)),
+    ok.
+
+all_accounts_balances(_Config) ->
+    rpc(application, set_env, [aehttp, enable_debug_endpoints, true]),
+    GenesisAccounts = rpc(aec_genesis_block_settings, preset_accounts, []),
+    BlocksToMine = 10,
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToMine),
+    {ok, 200, #{<<"accounts_balances">> := Balances}} = get_all_accounts_balances(),
+    {ok, MinerPubKey} = rpc(aec_keys, pubkey, []),
+    {ok, MinerBal} = rpc(aec_conductor, get_miner_account_balance, []),
+    ExpectedBalances = [{MinerPubKey, MinerBal} | GenesisAccounts],
+
+    true = length(Balances) =:= length(ExpectedBalances),
+    true =
+        lists:all(
+            fun(#{<<"pub_key">> := PKEncoded, <<"balance">> := Bal}) ->
+                Account = {base64:decode(PKEncoded), Bal},
+                lists:member(Account, ExpectedBalances) end,
+            Balances),
+    %% TODO: after fixing the aec_conductor mining issue, spend some txs
+    %% and check all accounts
+    ok.
+
+all_accounts_balances_empty(_Config) ->
+    rpc(application, set_env, [aehttp, enable_debug_endpoints, true]),
+    GenesisAccounts = rpc(aec_genesis_block_settings, preset_accounts, []),
+    {ok, 200, #{<<"accounts_balances">> := Balances}} = get_all_accounts_balances(),
+    true = length(Balances) =:= length(GenesisAccounts),
+    true =
+        lists:all(
+            fun(#{<<"pub_key">> := PKEncoded, <<"balance">> := Bal}) ->
+                Account = {base64:decode(PKEncoded), Bal},
+                lists:member(Account, GenesisAccounts) end,
+            Balances),
+    ok.
+
+all_accounts_balances_disabled(_Config) ->
+    rpc(application, set_env, [aehttp, enable_debug_endpoints, false]),
+    {ok, 404, _} = get_all_accounts_balances(),
+    ok.
+
+%% TODO: GetInfo: should it be disabled?
```

Comment:
I will create a specific pivotal story to address this.

Reply:
As discussed with @zp-sd 
* create `/version` for returning the version, revision and genesis
* protect /info and add more data to it if needed (block difficulties maybe?)
* change 404 with {} to smth more appropriate when debug flag is disabled

Reply:
[Pivotal story](https://www.pivotaltracker.com/story/show/153780478)